servicebus/data-plane/azure-servicebus/src/main/java/com/microsoft/azure/servicebus/primitives/ClientConstants.java

```diff
@@ -192,12 +194,22 @@ private ClientConstants() { }
     private static String getClientVersion() {
         String clientVersion;
         final Properties properties = new Properties();
+        InputStream clientPropInputStream = null;
         try {
-            properties.load(ClientConstants.class.getResourceAsStream("/client.properties"));
```

Comment:
[SpotBugs-P2]: com.microsoft.azure.servicebus.primitives.ClientConstants.getClientVersion() may fail to clean up java.io.InputStream


This method may fail to clean up (close, dispose of) a stream, database object, or other resource requiring an explicit cleanup operation. 
In general, if a method opens a stream or other resource, the method should use a try/finally block to ensure that the stream or resource is cleaned up before the method returns. 
This bug pattern is essentially the same as the OS_OPEN_STREAM and ODR_OPEN_DATABASE_RESOURCE bug patterns, but is based on a different (and hopefully better) static analysis technique. We are interested is getting feedback about the usefulness of this bug pattern. For sending feedback, check: 
contributing guideline
malinglist
In particular, the false-positive suppression heuristics for this bug pattern have not been extensively tuned, so reports about false positives are helpful to us. 
See Weimer and Necula, Finding and Preventing Run-Time Error Handling Mistakes, for a description of the analysis technique.